src/init.cpp

```diff
@@ -668,6 +668,7 @@ static void ThreadImport(std::vector<fs::path> vImportFiles)
 
     // -reindex
     if (fReindex) {
+        CBufferedFile blkdat(160*MAX_BLOCK_SERIALIZED_SIZE, 140*MAX_BLOCK_SERIALIZED_SIZE, SER_DISK, CLIENT_VERSION);
```

Comment:
This is a much larger buffer (640 mb), but it only exists during the first phase of reindexing (when you see `Reindexing block...` in the debug log); after that, the `bitcoind` memory usage increases for other reasons anyway. During the first phase, memory (`VSZ` in the ps output) is about 2.4 gb on my system; afterwards (when it's logging `UpdateTip: new best...`), it's about 3.5 gb. The benefit of a large buffer is that a larger fraction of the out-of-order blocks (that have to be processed later, when their parents are seen) can be found in memory, rather than having to read them again from disk. The "hit ratio" with this buffer size is just under 99%. The difference between 160 (but overall buffer size in max-blocks)) and 140 (the rewind limit) determines the size of disk reads (the larger, the better, more efficient).